### PR TITLE
Add support for Python 3.5, 3.6, PyPy2 and PyPy3.

### PR DESCRIPTION
**Pull request by icemac**
Drop support for Python 2.6 and 3.3.

### CODE REVIEW

File: src/martian/tests/test_all.py [hunk 1]
```diff
@@ -1,18 +1,33 @@
...
+from zope.testing import renormalizing
+
 from martian.testing import FakeModule
 from martian.testing import FakeModuleObject
 
-optionflags = doctest.NORMALIZE_WHITESPACE + doctest.ELLIPSIS + doctest.IGNORE_EXCEPTION_DETAIL
+
+optionflags = (doctest.NORMALIZE_WHITESPACE
+               + doctest.ELLIPSIS
+               + doctest.IGNORE_EXCEPTION_DETAIL)
```
Comment (mgedmin) [inline src/martian/tests/test_all.py]: Would mildly prefer `|` instead of `+`.

File: .travis.yml [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
-    - python bootstrap.py
-    - bin/buildout
+    - pip install -U pip setuptools
+    - pip install -U zope.testrunner coverage coveralls
+    - pip install -U -e .[test]
 script:
-    - bin/test -v1
+    - coverage run -m zope.testrunner --test-path=src
+after_success:
+    - coveralls
```
Comment (hugovk) [inline .travis.yml]: How about something like `pip install coveralls && coveralls` here, and remove `coveralls` from `install:`?

That way, if a build fails, there's no need to install unnecessary dependencies -> faster builds + fail fast.

**Comment by hugovk** [comment]
Not strictly related, but how about also changing README.txt into README.md and adding Travis CI and Coveralls badges?

**Comment by mgedmin** [comment]
> Not strictly related, but how about also changing README.txt into README.md and adding Travis CI and Coveralls badges?

In zopefoundationland we prefer README.rst ;)

File: src/martian/tests/test_all.py (hunk 1 above)
Comment (icemac) [reply src/martian/tests/test_all.py]: Adjusted.

File: .travis.yml (hunk 2 above)
Comment (icemac) [reply .travis.yml]: Interesting idea, lets see if it can buy us something.

**Comment by icemac** [comment]
@hugovk wrote:
> Not strictly related, but how about also changing README.txt into README.md and adding Travis CI and Coveralls badges?

I'll make this changes in a next PR.

File: .travis.yml (hunk 2 above)
Comment (icemac) [reply .travis.yml]: @hugovk Maybe this is not such a good idea: The build cache is created before running the `after_success` section so `coveralls` and its dependencies do not make it into this cache. I fear this makes successful builds some seconds slower. Maybe this does not matter, so I'll keep the change for now.

**Pull request merged**